bgp/gobgp/plugin.go

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2017 Pantheon technologies s.r.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//Package gobgp contains Ligato GoBGP BGP Plugin implementation
+package gobgp
+
+import (
+	"fmt"
+	"github.com/ligato/bgp-agent/bgp"
+	log "github.com/ligato/cn-infra/logging/logrus"
+	goBgpConfig "github.com/osrg/gobgp/config"
+	goBgpServer "github.com/osrg/gobgp/server"
+	"strconv"
+)
+
+// Plugin is GoBGP Ligato BGP Plugin implementation
+type Plugin struct {
+	SessionConfig    *goBgpConfig.Bgp
+	server           *goBgpServer.BgpServer
+	watcherCallbacks []func(*bgp.ReachableRoute)
+	stopWatch        chan bool
+	PluginName       string
+	Log              *log.Logger
+}
+
+//New creates a GoBGP Ligato BGP Plugin implementation
+func New() *Plugin {
+	return &Plugin{watcherCallbacks: []func(*bgp.ReachableRoute){}}
+}
+
+//Init uses plugin config and starts gobgp with dedicated goroutine for watching gobgp.
+func (plugin *Plugin) Init() error {
```

Comment:
please mention the comment two things:
- you are watching best path
- you are configuring neighbours (by the way what is the proper order of initializaion - add neighbours before start or before serve?)